src/menu/quicklinksMenu.js

```diff
@@ -51,4 +54,18 @@ function initializeEnabledSwitch() {
 		}
 	});
 }
-initializeEnabledSwitch();
\ No newline at end of file
+initializeEnabledSwitch();
+
+// Functionality for download lecture button.
+// Gets the url from storage and prompts a download.
+downloadLec.onclick = function() {
+	chrome.storage.sync.get('lecture', function(data) {
+		if (data.lecture) { // The download will not happen if the link is empty.
+			chrome.downloads.download({
+				url: data.lecture,
+				filename: 'lecture.mp4',
+				saveAs: true
+			});
```

Comment:
```suggestion
			chrome.permissions.request({permissions: ['downloads']}, function(granted) {
				if (granted) {
					chrome.downloads.download({
						url: data.lecture,
						filename: "last-viewed-lecture-recording.mp4",
						saveAs: true
					})
				} 
				else {
					alert('Download permission not granted.')
				}
			});
```
Here's where we would request the now optional downloads permission